src/main/java/seedu/address/logic/parser/BatchCommandParser.java

```diff
@@ -0,0 +1,164 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INSURANCE_POLICY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.BatchCommand;
+import seedu.address.logic.commands.DeleteCommand;
+import seedu.address.logic.commands.EditCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Handles all parsing of {@code BatchCommand} as well as arguments for {@code EditCommand} or {@code DeleteCommand}.
+ */
+public class BatchCommandParser implements Parser<BatchCommand> {
+    public static final String INVALID_BATCH_COMMAND = "Invalid batch operation!\nOnly edit and delete operations "
+            + "are supported.";
+    public static final String INVALID_EDIT_ARGUMENTS = "Invalid arguments for edit command!\nOnly tags and "
+            + "insurance policies can be edited in batch.";
+
+    /**
+     * Parses input to prepare for a {@code BatchCommand}, and the {@code Commands} that will be executed by it.
+     * First, parses and checks the validity of the {@code Command} that is to be executed in batch.
+     * Then, parses and checks the validity of the indices and arguments (if applicable) to be passed to the
+     * {@code Command} to be executed in bulk.
+     *
+     * @param args arguments of the {@BatchCommand} passed in by the user
+     * @return a {@code BatchCommand} with the {@code List} of {@code Commands} to be executed in batch.
+     * @throws ParseException if the user input for {@code BatchCommand} or the {@code Commands} does not conform to
+     *      the expected format.
+     */
+    public BatchCommand parse(String args) throws ParseException {
+        try {
+            String[] splitCommandAndIndicesAndArgs = args.trim().split(" ", 2);
+            String inputCommand = splitCommandAndIndicesAndArgs[0].trim();
+
+            // Checks the validity of the Command that the user passed as input to the BatchCommand
+            switch (inputCommand) {
+
+            case EditCommand.COMMAND_WORD:
```

Comment:
There could be a break here.

Reply:
I searched online what the coding standard should be, and it seems like I should have commented to say that this is a fallthrough case. I've provided the link here if you want to take a look. https://www.oracle.com/java/technologies/javase/codeconventions-statements.html